app/assets/javascripts/listings/ListingConstantsService.js.coffee

```diff
@@ -0,0 +1,103 @@
+############################################################################################
+####################################### SERVICE ############################################
+############################################################################################
+
+ListingConstantsService = () ->
+  Service = {}
+  Service._mohcdApplicationURLBase = 'http://sfmohcd.org/sites/default/files/Documents/MOH/BMR%20Rental%20Paper%20Applications/'
+  Service._mohcdEnglishApplicationURL = Service._mohcdApplicationURLBase + 'English%20BMR%20Rent%20Short%20Form%20Paper%20App.pdf'
+
+  Service.defaultApplicationURLs = [
+    # http://sfmohcd.org/general-bmr-rental-application
+    {
+      'language': 'English'
+      'label': 'English'
+      'url': Service._mohcdEnglishApplicationURL
+    }
+    {
+      'language': 'Spanish'
+      'label': 'EspaÃ±ol'
+      'url': Service._mohcdEnglishApplicationURL.replace('English', 'Spanish')
+    }
+    {
+      'language': 'Traditional Chinese'
+      'label': 'ä¸­æ–‡'
+      'url': Service._mohcdEnglishApplicationURL.replace('English', 'Chinese')
+    }
+    {
+      'language': 'Tagalog'
+      'label': 'Filipino'
+      'url': Service._mohcdEnglishApplicationURL.replace('English', 'Tagalog')
+    }
+  ]
+
+  Service.fieldsForUnitGrouping = [
+    'Unit_Type',
+    'Reserved_Type',
+    'BMR_Rent_Monthly',
+    'BMR_Rental_Minimum_Monthly_Income_Needed',
+    'Rent_percent_of_income',
+    'Status',
+  ]
+
+  Service.preferenceMap =
+    certOfPreference: "Certificate of Preference (COP)"
+    displaced: "Displaced Tenant Housing Preference (DTHP)"
+    liveWorkInSf: "Live or Work in San Francisco Preference"
+    liveInSf: "Live or Work in San Francisco Preference"
+    workInSf: "Live or Work in San Francisco Preference"
+    neighborhoodResidence: "Neighborhood Resident Housing Preference (NRHP)"
+    assistedHousing: "Rent Burdened / Assisted Housing Preference"
+    rentBurden: "Rent Burdened / Assisted Housing Preference"
+    antiDisplacement: "Anti-Displacement Housing Preference (ADHP)"
+    aliceGriffith: "Alice Griffith Housing Development Resident"
+
+  # Create a mapping to Salesforce naming conventions
+  Service.RESERVED_TYPES = {
+    VETERAN: 'Veteran'
+    DISABLED: 'Developmental disabilities'
+    SENIOR: 'Senior'
+  }
+
+  # TODO: -- REMOVE HARDCODED FEATURES --
+  Service.LISTING_MAP = {
```

Comment:
this is still being used, by `mapSlugToId`. i'm not fully certain what that whole setup is for. i recall that one reason for the use of `listingIs` and the `LISTING_MAP` is to check listing identity to apply special cases. cannot recall why a name/id mapping is used instead of directly checking a listing prop. may be partially to centralize all instances of special cases to make it easier to track them for future removal, since we do want to eventually remove all special cases. it may be worth looking at the git history around this to uncover what more `LISTING_MAP`, `listingIs`, and `mapSlugToId` are used for. i'm happy to do this as i have a little more historical context to apply. i'm inclined to put that work into a new story as it's not blocking this PR's functionality - what are your thoughts?

Reply:
I made a ticket going through to remove harcoded things https://www.pivotaltracker.com/story/show/163463841. We can just include that investigation when we do that ticket

Reply:
sounds good!